### PR TITLE
Update AsyncStorage doc

### PR DESCRIPTION
**Pull request by caabernathy**
Relates to #8203 for AsyncStorage API update.

- Added a small example to the intro section.
- Added jsdoc format tags to show up class description, parameter descriptions.
- Word-smithed many of the method descriptions.

I also made a bug fix to the autogen. It wasn't handling the scenario where a method may have no parameters.

**Test plan (required)**

Wrote a small sample app to test the snippet added to the intro section.

Ran website locally: http://localhost:8079/react-native/docs/asyncstorage.html

api_asyncstorage

Ran changed files through the linter.

### CODE REVIEW

**Comment by facebook-github-bot-5** [comment]
By analyzing the blame information on this pull request, we identified @geirman and @janicduplessis to be potential reviewers.

File: Libraries/Storage/AsyncStorage.js [hunk 1]
```diff
@@ -21,27 +22,58 @@ var RCTAsyncFileStorage = NativeModules.AsyncLocalStorage;
 var RCTAsyncStorage = RCTAsyncRocksDBStorage || RCTAsyncSQLiteStorage || RCTAsyncFileStorage;
 
 /**
- * AsyncStorage is a simple, asynchronous, persistent, key-value storage
+ * @class
```
Comment (JoelMarcey) [inline Libraries/Storage/AsyncStorage.js]: Just curious, why did you have to put `@class` here, but not in something like `AlertIOS` when you updated that?

File: website/layout/AutodocsLayout.js [hunk 2]
```diff
@@ -650,7 +650,7 @@ var Method = React.createClass({
           </span> || ''}
           {this.props.name}
           <span className="propType">
-            ({this.props.params
+            ({this.props.params && this.props.params.length && this.props.params
```
Comment (JoelMarcey) [inline website/layout/AutodocsLayout.js]: nice catch.

**Comment by JoelMarcey** [comment]
@facebook-github-bot shipit

**Comment by facebook-github-bot-0** [comment]
Thanks for importing. If you are an FB employee go to [Phabricator](https://our.intern.facebook.com/intern/opensource/github/pull_request/1735370346722304/int_phab) to review.

**Pull request closed without merging**